src/Compilers/Core/Portable/Diagnostic/LocalizableString.cs

```diff
@@ -45,29 +43,61 @@ public override bool Equals(object other)
             return Equals(other as LocalizableString);
         }
 
-        private sealed class FixedLocalizableString : LocalizableString
+        internal LocalizableString MakeExceptionSafe()
```

Comment:
I discussed this with Sri. We have 2 options, either to have a mutable field for OnException that can be hooked up later OR an immutable field for it. In the later case we are force to allocate a new LocalizableString for every LocalizableString passed into DiagnosticDescriptor constructor, where we are not mutating any state but just the OnException hander. In the former case, we can re-use existing instance of LocalizableString. First approach is better allocation wise, second one is better implementation and maintenance wise. I am not sure what is preferable, I hope people can give their opinions here and we can select the desired approach.